src/react-native/metro.ts

```diff
@@ -25,11 +25,100 @@ const b = recast.types.builders;
 
 const metroConfigPath = 'metro.config.js';
 
-export async function patchMetroConfig() {
+export async function patchMetroWithSentryConfig() {
   const mod = await parseMetroConfig();
 
   const showInstructions = () =>
-    showCopyPasteInstructions(metroConfigPath, getMetroConfigSnippet(true));
+    showCopyPasteInstructions(
+      metroConfigPath,
+      getMetroWithSentryConfigSnippet(true),
+    );
+
+  const success = await patchMetroWithSentryConfigInMemory(
+    mod,
+    showInstructions,
+  );
+  if (!success) {
+    return;
+  }
+
+  const saved = await writeMetroConfig(mod);
+  if (saved) {
+    clack.log.success(
+      chalk.green(`${chalk.cyan(metroConfigPath)} changes saved.`),
+    );
+  } else {
+    clack.log.warn(
+      `Could not save changes to ${chalk.cyan(
+        metroConfigPath,
+      )}, please follow the manual steps.`,
+    );
+    return await showInstructions();
+  }
+}
+
+export async function patchMetroWithSentryConfigInMemory(
+  mod: ProxifiedModule,
+  showInstructions: () => Promise<void>,
+): Promise<boolean> {
+  if (hasSentryContent(mod.$ast as t.Program)) {
+    const shouldContinue = await confirmPathMetroConfig();
+    if (!shouldContinue) {
+      await showInstructions();
+      return false;
+    }
+  }
+
+  const configExpression = getModuleExportsAssignmentRight(
+    mod.$ast as t.Program,
+  );
+  if (!configExpression) {
+    clack.log.warn(
+      'Could not find Metro config, please follow the manual steps.',
+    );
+    await showInstructions();
+    return false;
+  }
+
+  const wrappedConfig = wrapWithSentryConfig(configExpression);
+
+  const replacedModuleExportsRight = replaceModuleExportsRight(
+    mod.$ast as t.Program,
+    wrappedConfig,
+  );
+  if (!replacedModuleExportsRight) {
+    clack.log.warn(
+      'Could not replace module.exports right side, please follow the manual steps.',
```

Comment:
```suggestion
      'Could not automatically wrap the config export, please follow the manual steps.',
```

just a suggestion, not blocking!